nemoguardrails/library/domain_hallucination/actions.py

```
@@ -0,0 +1,397 @@
+
+"""NeMo Guardrails actions for domain hallucination detection."""
+
+from __future__ import annotations
+
+import logging
+import time
+from typing import Any, Dict, Optional
+
+try:  # pragma: no cover - available in NeMo runtime
+    from nemoguardrails.actions import action
+except Exception:  # pragma: no cover - standalone import fallback
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)